third_party/blink/renderer/platform/fonts/shaping/harfbuzz_shaper.cc

```diff
@@ -789,7 +789,16 @@ void HarfBuzzShaper::ShapeSegment(
 
       current_font_data_for_range_set =
           fallback_iterator.Next(fallback_chars_hint);
-      if (!current_font_data_for_range_set->FontData()) {
+
+      // [RecordReplay] See RUN-1302
+      // https://linear.app/replay/issue/RUN-1302#comment-0930eb4f
+      // When replaying, the `fallback_iterator.Next()` call can return
+      // null.  This is a hack to handle that case.
+      bool skipCheck =
+        recordreplay::IsReplaying() &&
```

Comment:
Should this maybe be HasDivergedFromRecording()?  When diverged we can get into weird situations like this.

Reply:
@bhackett1024 Is the font going to be there in other, non-diverged `UpdateLayout` calls?

Reply:
@Domiii we actually have a pretty elaborate system for dealing with font related info after diverging from the recording, see BindingFontsLinux.cpp (and other files for other platforms).  This is important for repainting and layout reflow, but it is still incomplete and the application can get into situations interacting with it that it normally wouldn't.  The font should be there before diverging from the recording (as long as it was originally there when recording), but when we diverge and *then* chrome/gecko etc. try to get font info from the system it might not be able to find everything it expects.

Reply:
I was looking through the driver/linker code and just noticed that a bunch of FontConfig and FreeType functions have CallPassThroughIfUnknown defined on them, and bindings innside the `BindingFonts*` code.  Makes sense.

Reply:
`CallPassThrough` might be pretty bad since font coverage is often notoriously different from system to system. 
E.g. our replay machines might not have standard arabic/chinese/japanese fonts etc. I imagine.